### PR TITLE
Add MyMoneyConsent component with radix checkbox

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
Add MyMoneyConsent component with radix checkbox

https://github.com/HedvigInsurance/racoon/assets/6661511/66807a16-4a81-42e7-802b-daf9d3137497

## Justify why they are needed
My Money porject

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
Current dependencies on/for this PR:
* **#3959** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3959?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3959?utm_source=stack-comment).

File: apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { useTranslation } from 'next-i18next'
+import React from 'react'
+import { CheckIcon, Text } from 'ui'
+import { checkboxIndicator, checkboxRoot, consentBody, consentWrapper } from './MyMoneyConsent.css'
+
+const MyMoneyConsent = () => {
+  const { t } = useTranslation('carDealership')
+  return (
+    <div className={consentWrapper}>
+      <Checkbox.Root className={checkboxRoot} defaultChecked id="myMoneyConsent">
```
Comment (alebedev) [inline apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx:11]: Meta-question - we were consistent with preferring switch over checkbox before. Is this intentional to use checkbox here?

File: apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.css.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  gridColumn: '2',
+})
+
+export const checkboxRoot = style({
+  display: 'flex',
+  placeContent: 'center',
+  backgroundColor: theme.colors.opaque1,
+  width: '1.5rem',
+  height: '1.5rem',
+  padding: '0.25rem',
```
Comment (alebedev) [inline apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.css.ts:26]: should we use theme sizes for things like .25rem, 1.5rem?

File: apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx (hunk 1 above)
Comment (alebedev) [inline apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx:11]: Good cal to not extract Checkbox as separate component yet!  Let's do it when we need to reuse

**Review by alebedev — APPROVED** [review]

File: apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx (hunk 1 above)
Comment (gustaveen) [reply apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx:11]: 🙌

File: apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.css.ts (hunk 2 above)
Comment (gustaveen) [reply apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.css.ts:26]: For padding definitely 👌

File: apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx (hunk 1 above)
Comment (gustaveen) [reply apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx:11]: Good question, it's intentional design decision to use a checkbox for consent in this case. Feels more formal than a switch and a common design pattern IMO

File: apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import * as Checkbox from '@radix-ui/react-checkbox'
+import { useTranslation } from 'next-i18next'
+import React from 'react'
+import { CheckIcon, Text } from 'ui'
+import { checkboxIndicator, checkboxRoot, consentBody, consentWrapper } from './MyMoneyConsent.css'
+
+const MyMoneyConsent = () => {
```
Comment (guilhermespopolin) [inline apps/store/src/features/carDealership/MyMoneyConsent/MyMoneyConsent.tsx:7]: We probably should keep the pattern of using named export for component.

```suggestion
export const MyMoneyConsent = () => {
```

**Review by guilhermespopolin — APPROVED** [review]
✅ Just left a minor comment.

**Comment by gustaveen** [comment]
### Merge activity

* **Feb 23, 8:57 AM EST**: @gustaveen merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3959).

**Pull request merged**